File: frontend/src/types/risk.ts
import { RiskLevel } from './contract';

export interface RiskFactor {
  name: string;
  score: number;
  weight: number;
  reason: string;
}

export interface RiskScore {
  id?: string;
  contract_id: string;
  overall_score: number;
  risk_level: RiskLevel;
  factors: RiskFactor[];
  created_at?: string;
}

export interface RiskScoreHistoryItem {
  id: string;
  overall_score: number;
  risk_level: RiskLevel;
  factors: RiskFactor[];
  created_at: string;
}

File: backend/app/schemas/risk.py
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel


class RiskFactor(BaseModel):
    name: str
    score: float
    weight: float
    reason: str


class RiskScoreResponse(BaseModel):
    id: Optional[str] = None
    contract_id: str
    overall_score: float
    risk_level: str
    factors: List[RiskFactor] = []
    created_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class RiskScoreHistoryItem(BaseModel):
    id: str
    overall_score: float
    risk_level: str
    factors: List[RiskFactor] = []
    created_at: datetime

    class Config:
        from_attributes = True
